packages/block-editor/src/components/spacing-sizes-control/input-controls/spacing-input-control.js

```diff
@@ -222,13 +224,26 @@ export default function SpacingInputControl( {
 						}
 						value={ currentValue }
 						units={ units }
-						min={ minimumCustomValue }
+						min={ minValue }
 						placeholder={ allPlaceholder }
 						disableUnits={ isMixed }
 						label={ ariaLabel }
 						hideLabelFromVision
 						className="spacing-sizes-control__custom-value-input"
 						size={ '__unstable-large' }
+						onDragStart={ () => {
+							if ( value?.charAt( 0 ) === '-' ) {
+								setMinValue( 0 );
+							}
+						} }
+						onDrag={ () => {
+							if ( value?.charAt( 0 ) === '-' ) {
+								setMinValue( 0 );
+							}
+						} }
```

Comment:
For what it's worth, I do think that it's better for UX if you only check for the sign on drag start. It would feel broken if you were dragging in the negative range and then it suddenly disallowed sub-zero dragging in the same drag gesture once you slip back a little over zero.

Reply:
I did this because then we would be able to drag from a positive to a negative, which is what we are trying to avoid. It would only jump to 0 if on drag stare we were already in the negatives

Reply:
The initial logic I suggested on Slack was like this, where the restriction would only kick in if the starting value was non-negative:

```diff
						onDragStart={ () => {
-							if ( value?.charAt( 0 ) === '-' ) {
+							if ( value?.charAt( 0 ) !== '-' ) {
								setMinValue( 0 );
							}
						} }
```

It is also more efficient because you won't need the `onDrag` handler, which is currently calling the setState many more times than necessary.

Reply:
I might have not explained myself. We are aiming to avoid people accidentally dragging to negative numbers, only allowing them if you are typing them directly in the input. Dragging should never let you get into the negatives

Reply:
Yes, I get that. My suggestion is specifically about what happens when the value was _already negative_ on drag start. I think you should be able to drag freely until the end of the drag gesture if the drag started from an already negative value.

That's just a UX intuition, so feel free to disregard if you feel otherwise!

Reply:
Perhaps one approach could be to check in `onDragStart` if the number began as negative when dragging starts and then store in state that we're allowing dragging through the range of negative values within `onDrag`? I.e. if when we start dragging we have a negative value, then allow dragging through any value as normal, and only set min value to `0` in `onDrag` if the value when we started dragging was non negative. To do that, I think we'd need another state variable for `valueWhenStartingToDrag`, though perhaps with a better name 😅

In any case, I think it could be worth tweaking in a follow-up, but in manual testing, the current state of this PR feels good to me.